src/Donations/Components/DonationsForm.tsx

```diff
@@ -187,6 +214,70 @@ function DonationsForm() {
     }
   }
 
+  const handlePlanetCashDonate = async () => {
+    const _onBehalfDonor = {
+      firstname: onBehalfDonor.firstName,
+      lastname: onBehalfDonor.lastName,
+      email: onBehalfDonor.email,
+    };
+
+    const _gift = {
+      ...giftDetails,
+      message: giftDetails.giftMessage,
+    };
+
+    delete _gift.giftMessage;
+
+    // create Donation data
+    const donationData = {
+      purpose: projectDetails!.purpose,
+      project: projectDetails!.id,
+      units: quantity,
+      prePaid: true,
+      onBehalf: onBehalf,
+      ...(onBehalf && { donor: _onBehalfDonor }),
+      ...(isGift && { gift: _gift }),
+    };
+
+    const cleanedDonationData = cleanObject(donationData);
+
+    const token = await getAccessTokenSilently();
+
+    // @method    POST
+    // @endpoint  /app/donation
+
+    try {
+      const { data, status } = await apiRequest({
+        url: "/app/donations",
+        method: "POST",
+        setshowErrorCard,
+        data: cleanedDonationData,
+        token,
+        headers: {
+          "IDEMPOTENCY-KEY": uuidv4(),
+        },
```

Comment:
From an software architecture point of view I would have preferred being able to switch on generating and sending an IDEMPOTENCY-KEY header by adding a kind of boolean option to our `apiRequest` for being able to use this feature also for other API calls later on.

Reply:
Done